### PR TITLE
megadriv.xml: compatibility field cleanups, apply minor fixes

### PR DESCRIPTION
**Pull request by angelosa**
- megadriv.xml: fix booger ROM loading;
- megadriv.xml: fix wboy5 and turmamon EEPROM saving;
- megadriv.xml: add SRAM to shijie, fixes roster names;
- megadriv.xml: allow nba2k and nba2k5 to boot;
- megacd.cpp: remove gross hack, allow megacdj & megacd2j to boot again;
- jcart.cpp: fix P3 and P4 labels;
- megadriv_acbl.cpp: make sonic3mb to boot, add basic protection sim;

### CODE REVIEW

**Comment by DavidHaywood** [comment]
for sracer you haver "Corrupted Ubi Soft and Vivid Image logos"

I'm not seeing this.

File: src/mame/machine/megadriv.cpp [hunk 1]
```diff
@@ -537,21 +555,21 @@ uint16_t md_base_state::megadriv_68k_check_z80_bus(offs_t offset, uint16_t mem_m
...
+		//LOG("%06x: 68000 check z80 Bus (byte LSB access) %04x\n", m_maincpu->pc(),mem_mask);
 		if (m_genz80.z80_has_bus || m_genz80.z80_is_reset) retvalue = 0x0001;
 		else retvalue = 0x0000;
 
 		return retvalue;
 	}
 	else
 	{
-		//logerror("%06x: 68000 check z80 Bus (word access) %04x\n", m_maincpu->pc(),mem_mask);
+		//LOG("%06x: 68000 check z80 Bus (word access) %04x\n", m_maincpu->pc(),mem_mask);
```
Comment (cuavas) [inline src/mame/machine/megadriv.cpp:572]: There’s no point converting these to `LOG` if you leave them commented out – they’ll still rot if they aren’t actually being parsed by the compiler.

**Review by cuavas — CHANGES_REQUESTED** [review]

File: src/mame/machine/megadriv.cpp [hunk 2]
```diff
@@ -28,21 +28,35 @@ Known Non-Issues (confirmed on Real Genesis)
 #include "speaker.h"
 
 
+#define VERBOSE (LOG_GENERAL)
```
[the remark this answers is not in the document]
Comment (angelosa) [reply src/mame/machine/megadriv.cpp:31]: It’s a generic logging, default driver doesn’t pass anything useful to debug issues. May turn it off but definitely need a smarter plan later on

**Comment by angelosa** [comment]
It's a lateral regression caused by masking irqs for sesame and fatalrew then. 
Was asking because a similar issue is happening with ncaafin4, and that **isn't** a regression but more likely something in DMA transfers that is not behaving properly.
Ditto for fatalrew in NTSC mode is not consistent at all, which can be a symptom of uninitialized variables, too.
Another culprit are the video timings themselves, I'm honestly surprised that the compatibility is unaffected by taking vblanks at line=186 and count hblanks every 1 usecs. Or even, it does for base HW, 32x and Mega CD likely cares more hence gives weird failures such as Sangokushi IV.

**Comment by angelosa** [comment]
Waiting for 5 full frames and shooting that irq asap once enabled isn't exactly what I call accurate either. 
For all I care on what's under the hood this is full on feasibility studies, and Warlock is barely mentioned with a `/* NO! (breaks warlock) */`. 
My head spins in trying to follow these files up, further bad wording will result in quick deletion of this branch and "not gonna look not even in 30 years" (cit.) because can't be bothered in trying to fix something that you seem eager to leave in current state with your name on top.

**Comment by DavidHaywood** [comment]
Well I'm saying, as the author of the current emulation, it's best nuked and rewritten.

Every last detail about the MD hardware is fully known at this point, these 15 year old guesses have no value.  Making new guesses today has no value, people know 100% how it all works, and that information should be used to rebuild it.

**Comment by angelosa** [comment]
Do you have a list of links that explains all these details so that I can add them on top of file(s)?

**Comment by DavidHaywood** [comment]
The most accurate reference is the Blast 'em code, although you can't use their code directly as it's GPL3, although I'm sure you can figure out hardware logic from it, which is factual.

Otherwise, get in contact with those doing the FPGA cores for Mister etc.

I don't know if there's a central repository for the information.

File: src/devices/bus/megadrive/jcart.cpp [hunk 3]
```diff
@@ -90,24 +90,24 @@ void md_seprom_mm96_device::device_add_mconfig(machine_config &config)
...
-	PORT_BIT( 0x0008, IP_ACTIVE_LOW, IPT_JOYSTICK_RIGHT ) PORT_8WAY PORT_PLAYER(4)
+	PORT_BIT( 0x0001, IP_ACTIVE_LOW, IPT_JOYSTICK_UP ) PORT_8WAY PORT_PLAYER(4) PORT_NAME("P4 Right")
+	PORT_BIT( 0x0002, IP_ACTIVE_LOW, IPT_JOYSTICK_DOWN ) PORT_8WAY PORT_PLAYER(4) PORT_NAME("P4 Right")
+	PORT_BIT( 0x0004, IP_ACTIVE_LOW, IPT_JOYSTICK_LEFT ) PORT_8WAY PORT_PLAYER(4) PORT_NAME("P4 Right")
+	PORT_BIT( 0x0008, IP_ACTIVE_LOW, IPT_JOYSTICK_RIGHT ) PORT_8WAY PORT_PLAYER(4) PORT_NAME("P4 Right")
 	PORT_BIT( 0x0010, IP_ACTIVE_LOW, IPT_BUTTON2 ) PORT_PLAYER(4) PORT_NAME("P4 B")
 	PORT_BIT( 0x0020, IP_ACTIVE_LOW, IPT_BUTTON3 ) PORT_PLAYER(4) PORT_NAME("P4 C")
 	PORT_BIT( 0x0040, IP_ACTIVE_LOW, IPT_BUTTON1 ) PORT_PLAYER(4) PORT_NAME("P4 A")
-	PORT_BIT( 0x0080, IP_ACTIVE_LOW, IPT_START ) PORT_PLAYER(4)
+	PORT_BIT( 0x0080, IP_ACTIVE_LOW, IPT_START ) PORT_PLAYER(4) PORT_NAME("P4 Start")
```
[the remark this answers is not in the document]
Comment (angelosa) [reply src/devices/bus/megadrive/jcart.cpp]: Works if I just do `PORT_PLAYER(1)`, `PORT_PLAYER(2)` and `%p`. 
Also fixed that P4 Right repeated four times whoops.

**Comment by angelosa** [comment]
I'm gonna revert what I've done with the pending mechanism for now, but I'll definitely centralize the interrupt mechanisms at some point because I already have horror stories about debugging this. 
A full rewrite is unfeasible given the ~90% compatibility, I'd rather get there with multiple feature branches instead.

**Comment by DavidHaywood** [comment]
> A full rewrite is unfeasible given the ~90% compatibility, 

I still think a full rewrite is the best option.  The ~90% compatibility is a stack of things done wrong that just happen to work together.  There's nothing *correct* about my old code at all.

**Comment by angelosa** [comment]
Are you volunteering to do it? (It’s a rhetorical question).

**Comment by angelosa** [comment]
> No, but I'm wondering why you won't accept my own admission that the code I wrote is pure trash and beyond a few fixups if you're ever wanting to do more than juggling compatibility between different sets of games.

Because the alternative is not feasible for the _end user_ , giving them confusion and giving even less willingness to update to latest. Also, I still don't have the full picture atm, so experimenting with a tracer bullet approach is much better than aimlessly full rewrite with a couple hours of regtesting per country to take into account. Picking up those pieces for an eventual "full" rewrite can happen later on anyway.
Examples of tracer bullets on top of my head: 

- Remove code duplication?
- Making it less of a chore to debug?
- Use actual screen timings instead of two approaches x 5 functions that are a nightmare to handle being scattered across different files?
- Make DMA closer to reality?
- Fix sprite masking?

All questions that I can answer by isolating individually and _intentional omissis_, and again: if you are not actively gonna code anything then all this preaching about what's **your** best approach is literally a blatant and unnecessary trolling attempt, hopefully the thread lockup won't get me another silly answer.

**Comment by angelosa** [comment]
Ready to merge once CI passes.

**Pull request merged**